framework/src/base/FEProblem.C

```diff
@@ -3459,10 +3459,14 @@ FEProblem::computeResidual(NonlinearImplicitSystem &/*sys*/, const NumericVector
   {
     computeResidualType(soln, residual, _kernel_type);
   }
-  catch(MooseException & e)
-  {
-    mooseError("Impossible");
```

Comment:
Why was this ever like this?  I think that the issue is that we weren't expecting a `MooseException` to get re-thrown from inside `computeResidualType()`.  It should be dealt with in there (set up the solver to fail like you note) and then swallowed there.

But maybe it's fine to swallow it here.